### ISSUE TITLE
Loading Random Map Bug

### ISSUE THREAD

**Issue by MattCollyer** [smoldata/smol-maps#13]
There seems to be a bug when Smol Maps loads a new random map.  Console says it cannot read 'active' property from data.map, which is undefined. Adding "active" property on line 297 in smol-maps.js (back-end) does not work...unless on the front end (smol.maps.js) on line 237 you change data.map.active to data.active  However, this still will not work because data.map is needed much  throughout the code. So, the issue? When loading a random map, data.map is undefined. It should be an easy fix, however I don't know what data.map should be set as, so a tricky fix on my end. 
tldr; add property "map" to "data"

**Comment by dphiffer** [comment]
Yeah, looks like there's some inconsistency in how the server reports to the front-end.

I feel like the front-end has it wrong here...

**Comment by dphiffer** [comment]
Turns out the backend had it wrong!

**Comment by dphiffer** [comment]
At the high level, what happened here is there are two functions, `load_map` and `save_map`. They work kind of like a wiki, if you try to load a map that doesn't exist yet, it saves a new one and pretends it was there all along.

*However*, for this to work properly the two functions need to offer the same kind of response to the client, which they didn't. I took an approach where `load_map` calls `save_map`, and passes a callback function for what to do when `save_map` finishes its business. The call back is just a second call to `load_map`, which picks up the newly-saved map.

Another approach that is a little more work, but probably a *better* solution, is to make `save_map` respond in exactly the same way as `load_map` does. The reason it's harder is that it just means tracking down all the places that call `save_map` on the front-end and bring them into line with `load_map` style.

For comparison, when you save a map, you get this:
```
{
    "ok": 1,
    "data": { (map data goes here) }
}
```

When you load a map, it includes the map and the venues associated with it:
```
{
    "ok": 1,
    "map": { (map data goes here) },
    "venues": [ (venues data goes here) ]
}
```

I think this interim fix is fine, we can always come back and clean this up later.
